Gemfile

```diff
@@ -10,9 +10,14 @@ def location_for(place, fake_version = nil)
   end
 end
 
-gem "puppet", *location_for(ENV['PUPPET_LOCATION'] || '~> 3.7.0')
-gem "facter", *location_for(ENV['FACTER_LOCATION'] || '~> 2.0')
-gem "hiera", *location_for(ENV['HIERA_LOCATION'] || '~> 1.0')
+if puppetversion = ENV['PUPPET_GEM_VERSION']
+  gem 'puppet', puppetversion,  :require => false
+else
+  gem 'puppet', *location_for(ENV['PUPPET_LOCATION'] || '~> 3')
+end
+
+gem "facter", *location_for(ENV['FACTER_LOCATION'] || '>= 1')
```

Comment:
It's mostly handled with the Puppet Gem install, but when you are faking the locations, Facter is not automatically pulled in as a gem. Same with Hiera.